command/agent/alloc_endpoint_test.go

```diff
@@ -261,31 +262,77 @@ func TestHTTP_AllocQuery_Payload(t *testing.T) {
 
 func TestHTTP_AllocStats(t *testing.T) {
 	t.Parallel()
+	require := require.New(t)
+
 	httpTest(t, nil, func(s *TestAgent) {
-		// Make the HTTP request
-		req, err := http.NewRequest("GET", "/v1/client/allocation/123/foo", nil)
-		if err != nil {
-			t.Fatalf("err: %v", err)
+		// Local node, local resp
+		{
+			// Make the HTTP request
+			req, err := http.NewRequest("GET", fmt.Sprintf("/v1/client/allocation/%s/stats", uuid.Generate()), nil)
+			if err != nil {
+				t.Fatalf("err: %v", err)
+			}
+			respW := httptest.NewRecorder()
+
+			// Make the request
+			_, err = s.Server.ClientAllocRequest(respW, req)
+			require.NotNil(err)
+			require.Contains(err.Error(), "unknown allocation")
 		}
-		respW := httptest.NewRecorder()
 
-		// Make the request
-		_, err = s.Server.ClientAllocRequest(respW, req)
-		if !strings.Contains(err.Error(), resourceNotFoundErr) {
-			t.Fatalf("err: %v", err)
+		// Local node, server resp
+		{
+			srv := s.server
+			s.server = nil
+
+			req, err := http.NewRequest("GET", fmt.Sprintf("/v1/client/allocation/%s/stats", uuid.Generate()), nil)
+			require.Nil(err)
+
+			respW := httptest.NewRecorder()
+			_, err = s.Server.ClientAllocRequest(respW, req)
+			require.NotNil(err)
+			require.Contains(err.Error(), "unknown allocation")
+
+			s.server = srv
+		}
+
+		// no client, server resp
+		{
+			c := s.client
+			s.client = nil
+
+			testutil.WaitForResult(func() (bool, error) {
+				n, err := s.server.State().NodeByID(nil, c.NodeID())
+				if err != nil {
+					return false, err
+				}
+				return n != nil, nil
+			}, func(err error) {
+				t.Fatalf("should have client: %v", err)
+			})
+
+			req, err := http.NewRequest("GET", fmt.Sprintf("/v1/client/allocation/%s/stats", uuid.Generate()), nil)
+			require.Nil(err)
+
+			respW := httptest.NewRecorder()
+			_, err = s.Server.ClientAllocRequest(respW, req)
+			require.NotNil(err)
+			require.Contains(err.Error(), "unknown allocation")
+
+			s.client = c
```

Comment:
another test case for no server, not a local node so local client can't handle the call?